### PR TITLE
add otel.instrumentation.http.server.add-server-attributes

### PR DESCRIPTION
**Pull request by zeitlinger**
Background:

- exploring different options for making it easier to opt-in to `server.address` and `server.port` for `http.server.request.duration`
- these 2 attributes are special in that they allow the creation of a service graph - for services that don't send spans
- the best opt-in mechanism should be implemented in all languages

Alternatives:

- waiting for file config to be available
  - can configure views that control attributes to include
  - have to mention all attributes, not just opt-in
- add the same option to instrumentation
  - seemed complicated to do - but will get easier due to https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/11605
  - tried for python: https://github.com/open-telemetry/opentelemetry-python-contrib/pull/2597
- add a more generic option `otel.metrics.opt-in=http.server.request.duration=server.address;server.port`
  - separator `;` is a bit odd, but `,` is already used to have entries for multiple metrics 
- create a distribution
  - feels excessive

### CODE REVIEW

**Comment by trask** [comment]
can you explain what problem this is solving? thanks

File: sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/AutoConfiguredOpenTelemetrySdkBuilder.java [hunk 1]
```diff
@@ -448,6 +453,25 @@ public AutoConfiguredOpenTelemetrySdk build() {
 
       if (sdkEnabled) {
         SdkMeterProviderBuilder meterProviderBuilder = SdkMeterProvider.builder();
+        if (config.getBoolean("otel.instrumentation.http.server.add-server-attributes", false)) {
```
Comment (jkwatson) [inline sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/AutoConfiguredOpenTelemetrySdkBuilder.java:456]: I don't think this belongs in our core SDK autoconfigure. This looks to be an instrumentation-specific option. Let's talk about the right place for this configuration in a SIG meeting.

**Comment by jack-berg** [comment]
We've added the first instrumentation configuration options in file configuration here: https://github.com/open-telemetry/opentelemetry-configuration/pull/91

Separately, this does not belong as a core SDK configuration.

**Comment by jkwatson** [comment]
@zeitlinger is there any reason to keep this open, given that both maintainers do not believe this is appropriate to build into the core repo?

**Comment by zeitlinger** [comment]
No

**Pull request closed without merging**